aospy/data_loader.py

```diff
@@ -0,0 +1,180 @@
+"""aospy DataLoader objects"""
+import xarray as xr
+from glob import glob
+
+from . import internal_names
+from .timedate import TimeManager
+
+
+class DataLoader(object):
+    """A fundamental DataLoader object"""
+    def __init__(self):
+        self.correction = None
+
+    def load_variable(self, var=None, start_date=None, end_date=None,
+                      intvl_in=None, vert_in=None):
+        """Return a DataArray with requested variable, given time range,
+        and input interval.
+
+        This function automatically renames all grid attributes to match
+        aospy conventions.
+
+        Parameters
+        ----------
+        var : Var
+             aospy Var object
+        start_date : netCDF4.netcdftime or np.datetime64
+             start date for interval
+        end_date : netCDF4.netcdftime or np.datetime64
+             end date for interval
+        intvl_in : str
+             interval coding (e.g. 'monthly')
+
+        Returns
+        -------
+        da : DataArray
+             DataArray for the specified variable, date range, and interval in
+        """
+        file_set = self._generate_file_set(var, start_date, end_date, intvl_in,
+                                           vert_in)
+        ds = xr.open_mfdataset(file_set, preprocess=self.rename_grid_attrs,
+                               concat_dim=internal_names.TIME_STR,
+                               decode_cf=False)
+        ds = TimeManager.enforce_valid_timestamp_date_range(ds)  # Tested
```

Comment:
Put these three lines into their own internal method, and potentially add the `xr.decode_cf` step here if (1) it doesn't mess things up to put that before the `self.set_grid_attrs_as_coords` step and (2) `decode_cf` doesn't modify anything besides the time-related quantities.